src/targets/pubDev.ts

```diff
@@ -0,0 +1,208 @@
+  }
+
+  /**
+   * Uploads all files to pub.dev using Dart CLI.
+   *
+   * @param version New version to be released
+   * @param revision Git commit SHA to be published
+   */
+  public async publish(_version: string, revision: string): Promise<any> {
+    await this.createCredentialsFile();
+
+    await withTempDir(
+      async directory => {
+        await this.cloneRepository(this.githubRepo, revision, directory);
+        await forEachChained(this.pubDevConfig.packages, async pkg =>
+          this.publishPackage(directory, pkg)
+        );
+      },
+      true,
+      'craft-pub-dev-'
+    );
+  }
+
+  public async createCredentialsFile(): Promise<void> {
+    const credentialsFilePath = this.getCredentialsFilePath();
+    const content = {
+      accessToken: this.pubDevConfig.PUBDEV_ACCESS_TOKEN,
+      refreshToken: this.pubDevConfig.PUBDEV_REFRESH_TOKEN,
+      tokenEndpoint: 'https://accounts.google.com/o/oauth2/token',
+      scopes: ['openid', 'https://www.googleapis.com/auth/userinfo.email'],
+      expiration: 1645564942000, // Expiration date is required, but irrelevant
+    };
+
+    // Store credentials file only if they doesn't exist
+    try {
+      await fsPromises.access(credentialsFilePath, constants.F_OK);
+      this.logger.warn('Credentials file already exists. Skipping creation.');
+    } catch {
+      await fsPromises.writeFile(credentialsFilePath, JSON.stringify(content));
+      this.logger.info('Credentials file created.');
+    }
+  }
+
+  private getCredentialsFilePath(): string {
+    const currentPlatform = platform();
+    switch (currentPlatform) {
+      case 'darwin':
+        return `${homedir()}/Library/Application Support/dart/pub-credentials.json`;
+      case 'linux':
+        return `${homedir()}/.config/dart/pub-credentials.json`;
+      default:
+        throw new Error(`Unsupported platform: ${currentPlatform}`);
+    }
+  }
+
+  /**
+   * Clones a repository.
+   *
+   * @param config Git configuration specifying the repository to clone.
+   * @param revision The commit SHA that should be checked out after the clone.
+   * @param directory The directory to clone into.
+   */
+  public async cloneRepository(
+    config: GitHubGlobalConfig,
+    revision: string,
+    directory: string
+  ): Promise<any> {
+    const { owner, repo } = config;
+    const git = simpleGit(directory);
+    const url = `https://github.com/${owner}/${repo}.git`;
+
+    this.logger.info(`Cloning ${owner}/${repo} into ${directory}`);
+    await git.clone(url, directory);
+    await git.checkout(revision);
+  }
+
+  /**
+   * Publishes a package on pub.dev
+   *
+   * @param directory The path to the root package
+   * @param package The path to the package itself, relative to the root
+   * @returns A promise that resolves when the package has been published
+   */
+  public async publishPackage(directory: string, pkg: string): Promise<void> {
+    // `--force` prevents confirmation prompt
```

Comment:
Is this confirmation a different step from the [URL authorization](https://github.com/getsentry/craft/pull/353/files#diff-faa5992b8b30b35d111a2cf7c9597ce8dd6b6b4fbc176987d5769b5c812294d8R199) below?